app/models/condition.rb

```diff
@@ -293,4 +297,42 @@ def self.import_from_hash(condition, options={})
 
     return c, status
   end
+
+  protected
+
+  module SafeNamespace
+    def self.eval_script(script, context)
+      _log.debug("Context: [#{context}], Class: [#{context.class.name}]")
+      _log.debug("Script:\n#{script}")
+      begin
+        t = Thread.new do
+          Thread.current["result"] = _eval(context, script)
+        end
+        to = 20 # allow 20 seconds for script to complete
+        Timeout::timeout(to) { t.join }
+      rescue TimeoutError => err
+        t.exit
+        _log.error  "The following error occurred during ruby evaluation"
+        _log.error  "  #{err.class}: #{err.message}"
+        raise "Ruby script timed out after #{to} seconds"
+      rescue Exception => err
+        _log.error  "The following error occurred during ruby evaluation"
+        _log.error  "  #{err.class}: #{err.message}"
+        raise "Ruby script raised error [#{err.message}]"
+      ensure
+        (t["log"] || []).each {|m| _log.info("#{m}")} unless t.nil?
+      end
+      return t["result"]
+    end
+
+    def self._eval(context, script)
+      proc { $SAFE = 3; eval(script) }.call
```

Comment:
I added the $SAFE=3 here to prevent expressions containing shell commands like `ls /` from being evaluated. The original code did not have this before is was removed.